numpy/core/tests/test_ufunc.py

```diff
@@ -1984,3 +1984,27 @@ def test_ufunc_noncontiguous(ufunc):
                 assert_allclose(res_c, res_n, atol=tol, rtol=tol)
             else:
                 assert_equal(c_ar, n_ar)
+
+
+@pytest.mark.parametrize('ufunc', [np.sign, np.equal])
+def test_ufunc_warn_with_nan(ufunc):
+    # issue gh-15127
+    # test that calling certain ufuncs with a non-standard `nan` value does not
+    # emit a warning
+    a = np.packbits([0, 0, 0, 0, 0, 0, 0, 1,
+                     0, 0, 0, 0, 0, 0, 0, 0,
+                     0, 0, 0, 0, 0, 0, 0, 0,
+                     0, 0, 0, 0, 0, 0, 0, 0,
+                     0, 0, 0, 0, 0, 0, 0, 0,
+                     0, 0, 0, 0, 0, 0, 0, 0,
+                     1, 1, 1, 1, 0, 0, 0, 0,
+                     0, 1, 1, 1, 1, 1, 1, 1])
```

Comment:
In my opinion this would be a lot clearer as:
```
np.array(0x7ff0000000000001, '<i8').view('<f8')
```
`packbits` has very weird default endiannes choices.

Reply:
That looks good to me.

According to [wikipedia](https://en.wikipedia.org/wiki/Endianness#Floating_point), *"...on modern standard computers (i.e., implementing IEEE 754), one may in practice safely assume that the endianness is the same for floating-point numbers as for integers..."*  So to create a native-endian signaling nan, it should be possible to drop the explicit endianess character `>` in that expression and write:

    b = np.array(0x7ff0000000000001, 'i8').view('f8')

I implemented that in my last update, and the tests still pass.